x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectRealm.java

```diff
@@ -461,18 +477,15 @@ static ClaimParser forSetting(Logger logger, OpenIdConnectRealmSettings.ClaimSet
                         "OpenID Connect Claim [" + claimName + "] for [" + setting.name(realmConfig) + "]",
                         claims -> {
                             Object claimValueObject = claims.getClaim(claimName);
+                            validateClaimType(claimValueObject, RealmSettings.getFullSettingKey(realmConfig, setting.getClaim()));
                             if (claimValueObject == null) {
                                 return List.of();
                             } else if (claimValueObject instanceof String) {
                                 return List.of((String) claimValueObject);
-                            } else if (claimValueObject instanceof List == false) {
-                                throw new SettingsException("Setting [" + RealmSettings.getFullSettingKey(realmConfig, setting.getClaim())
-                                    + " expects a claim with String or a String Array value but found a "
-                                    + claimValueObject.getClass().getName());
                             }
                             return ((List<String>) claimValueObject).stream()
```

Comment:
The new `validateClaimType` method does not assert `claimValueObject` is a `List`. In theory it could be a `Set` or a `Number`.

Reply:
`validateClaimType` right above validates that is either a Collection or a String. Then, if this is not a String, it is a Collection. How could it get to be a Set or a Number ? Maybe I didn't get what you mean